splice_spark/src/main/spark2.2/com/splicemachine/spark/splicemachine/SplicemachineContext.scala

```diff
@@ -24,62 +24,44 @@ import com.splicemachine.derby.stream.spark.SparkUtils
 import com.splicemachine.derby.vti.SpliceDatasetVTI
 import com.splicemachine.derby.vti.SpliceRDDVTI
 import com.splicemachine.tools.EmbedConnectionMaker
+import org.apache.log4j.Logger
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.execution.datasources.jdbc._
 import org.apache.spark.sql.jdbc.{JdbcDialect, JdbcDialects, JdbcType}
 import org.apache.spark.sql.types._
 import org.apache.spark.sql.{DataFrame, Dataset, Row}
-import org.apache.spark.SerializableWritable
 import java.util.Properties
 
 import com.splicemachine.access.HConfiguration
-import com.splicemachine.access.hbase.HBaseConnectionFactory
 import org.apache.hadoop.fs.{FileSystem, Path}
 import org.apache.hadoop.fs.permission.FsPermission
 import org.apache.hadoop.hbase.security.token.AuthenticationTokenIdentifier
 import org.apache.hadoop.io.Text
-import org.apache.hadoop.security.{Credentials, UserGroupInformation}
-import org.apache.spark.broadcast.Broadcast
+import org.apache.hadoop.security.UserGroupInformation
 import org.apache.hadoop.security.token.Token
-import org.apache.hadoop.hbase.security.token.TokenUtil
 import org.apache.hadoop.hbase.util.Bytes
-import org.apache.log4j.Logger
 import org.apache.spark.api.java.JavaRDD
 import org.apache.spark.scheduler.{SparkListener, SparkListenerApplicationEnd}
 
-object Holder extends Serializable {
-  @transient lazy val log = Logger.getLogger(getClass.getName)
```

Comment:
Keep the logger here in the Holder object, that way it's static rather than local to each SpliceMachineContext